pkg/crypto/edg.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package crypto
+
+import "crypto/ecdsa"
+
```

Comment:
if we wanted to have explicit check if these structures implements `keystore.EDG` interface here via construct:

```
var _ keystore.EDG = (*Secp256k1EDG)(nil)
var _ keystore.EDG = (*Secp256r1EDG)(nil)
```
it would not be possible because this would lead to circular imports, right? 

If this is the case would it make more sense to have these structs specified in `keystore` package? Maybe have `edg.go` file which will hold interface together with these structures?

Or maybe `EDG` interface is more suitable in crypto package?

---

Ether way we decide it would be nice to have this explicit check because it further engraves design.

Reply:
added types

Reply:
I am not sure if `crypto` package should have dependency to `keystore` package. `crypto` package should be at higher level of abstraction then `keystore` therefore is shouldn't depend on it.